### ISSUE TITLE
Failed to save event

### ISSUE THREAD

**Issue by Sniper26** [AnonymousController/wstc#1]
Been using this script for awhile since VT went away however long ago. It has been great! However, all of a sudden with working mid shifts I get this error pretty often. Not entirely sure how I can go about fixing it. Generally, the script will put work well and put mid shifts correctly, but it seems some weeks it errors out. Any help would be much appreciated as honestly this makes checking my schedule so much easier. Google Calendar export and being able to share that calendar without a lot of manual input has been fantastic. Thank you
<img width="424" alt="Screen Shot 2021-01-30 at 12 51 12 PM" src="https://user-images.githubusercontent.com/1342927/106367806-39713a80-62fa-11eb-9984-4dc0f92c4345.png">

**Comment by AnonymousController** [comment]
Strange. My "sanity check" for event times is pretty simple:

    if (evtEndDate < evtStartDate) then set evtEndDate to (evtEndDate + 1 * days)

So in theory that error should never happen... are you still seeing it every so often? There *shouldn't* be more than one event per day shown in WMT Scheduler, at least I don't think so. For this 2230 shift, what is the date according to WMT? And is there another shift the previous date, or the same date?

**Comment by Sniper26** [comment]
Even with the new build, something is happening. Weird. I'll go in and see if I can mess with the code itself as well. Thanks!

**Comment by AnonymousController** [comment]
My best guess is that it is something to do with the "date" (rather than time) being funky for mid shifts. How is it shown in WMT Scheduler? Is the mid shift on a separate date? Is it true that there are some calendar days (0000-2359L) on which two separate shifts begin?

**Comment by Sniper26** [comment]
Interesting. This is my only facility so I wasn't aware that the days that are shown for mids would show differently at other facilities. For example, if I go to a single day on Webschedule, the day "starts" with the people on the mid coming in at 2230. Even though that 2230 would technically start the day prior.
![one day](https://user-images.githubusercontent.com/1342927/127384433-fda39807-8bff-493e-ad97-3ec355839612.png)



It would make sense to me that it would try and make the shift start at 2230 on that day, even though it would technically be the day before. What I was doing prior was just moving the shift one day in my calendar before the error began. Either way, thank you for all the work! Makes my life so much easier. Now if they would update webschedule to be useful.... one can dream.

**Comment by AnonymousController** [comment]
Okay. So if you see 2230 shift that appears on Web Scheduler as being "on Friday," you would actually work your morning shift on Thursday and then come back Thursday night at 2230 for your "Friday" mid shift? I can see how that works, and I can definitely see how it would cause problems with my code. Though it still should be able to create the event, it would just show the event going from Friday night into Saturday... hm. I haven't heard from other people with the issue, so I don't know if mids are indicated differently at their facilities or if they just haven't complained to me.

**Comment by Sniper26** [comment]
Exactly, you got it. I'll mess with it some more and see if I can see why it isn't creating an event going Friday to Saturday any longer. That is what it was doing prior. It was an easy thing to get around as I just would move that even manually. Still so much better than having nothing. Either way, appreciate the responses and work!

**Comment by AnonymousController** [comment]
@Sniper26 the script now creates events for midnight shifts in a way that matches reality. Does this solve your issue or no?
